src/compiler/generate-common-tests.xsl

```diff
@@ -302,7 +302,7 @@
    <xsl:template match="x:variable" mode="x:generate-calls">
       <xsl:param name="vars" select="()" tunnel="yes" as="element(var)*"/>
       <!-- The variable declaration. -->
-      <xsl:if test="empty(following-sibling::x:call)">
+      <xsl:if test="empty(following-sibling::x:call) and empty(following-sibling::x:context)">
          <xsl:apply-templates select="." mode="test:generate-variable-declarations">
             <xsl:with-param name="var"  select="( @name, generate-id() )[1]"/>
```

Comment:
Just curious. What is this `generate-id()` used for?

Reply:
I'm not 100% sure because it predated my work. A guess is that someone copied the code from the `match="x:param" mode="x:compile"` template. In that template, the expression serves a purpose because `x:call/x:param` does not require a name. In this template, I don't see the point in supporting an anonymous variable, because you need a stable name in order to reference the variable. If you think I should change the attribute on this line to `select="@name"`, let me know.

Reply:
I agree to your guess. Let's take this opportunity to fix it:

```suggestion
            <xsl:with-param name="var"  select="@name"/>
```